proxy/ControlBase.cc

```diff
@@ -114,19 +113,19 @@ TimeMod::check(HttpRequestData *req) const
 TimeMod *
 TimeMod::make(char *value, const char **error)
 {
-  Tokenizer rangeTok("-");
-  TimeMod *mod = nullptr;
-  TimeMod tmp;
-  int num_tok;
-
-  num_tok = rangeTok.Initialize(value, SHARE_TOKS);
-  if (num_tok == 1) {
+  TimeMod *mod    = nullptr;
+  char *range_end = std::strchr(value, '-');
+  if (nullptr == range_end) {
     *error = "End time not specified";
-  } else if (num_tok > 2) {
-    *error = "Malformed time range";
-  } else if (nullptr == (*error = timeOfDayToSeconds(rangeTok[0], &tmp.start_time)) &&
-             nullptr == (*error = timeOfDayToSeconds(rangeTok[1], &tmp.end_time))) {
-    mod = new TimeMod(tmp);
+  } else {
+    *(range_end++) = '\0';
```

Comment:
I used strchr() instead of TextView due to the need to poke in null chars for compatibility with C-style utils.